### DISCUSSION TITLE
Management of Revisions and Drafts Across Sections

### DISCUSSION THREAD

**Discussion by richhayler** [craftcms/cms#8711] [Ideas]
### Description
I'm requesting some consideration to equip site admins with tools to purge entry revisions and drafts. Currently the only way to do this within the CP is by selecting the individual entry and then draft or revision and then selecting to delete it.

I'm encountering instances where there are multiple drafts for an entry where there are no differences between each draft but they are the result of an author using live preview and then clicking back out of the entry to the entry index.

While there is a config setting to set the max # of revisions it would be great to have access to CP utilities to delete all revisions either globally or by selecting applicable sections and to do the same for drafts.

**Comment by ccsantos23** [comment]
It would be nice if hitting preview would send the author to his existing draft, instead of creating a new one.
And also have drafts invalidated or marked "stale" when they become older than the entry's current revision.

**Comment by jwb009** [comment]
I agree, the system as-is is confusing to authors. Drafts should not be created if no changes have been made while in Live Preview. Our authors also get confused that two new icons appear -- the checkmark and pencil icon -- after they leave Live Preview.

**Comment by ghost** [comment]
2 things that came up (besides #4471 ...)

* You should not be able to create a conflicting draft without notice, so if a draft already exists, there should always be a warning/confirmation when clicking either 'create draft' or 'preview' from the source entry.
* I get the technical issue behind silently creating a draft for preview, but for better ux there could be a message and an undo beneath the 'Close Preview' button. Something like 'A draft has been created (to enable preview)' [Undo]' , where the undo button deletes the new draft and returns to the source entry.
